ksqldb-tools/src/test/java/io/confluent/ksql/tools/migrations/MigrationsTest.java

```diff
@@ -108,42 +134,74 @@ public void testApply() throws IOException {
         "CREATE STREAM BAR (A STRING) WITH (KAFKA_TOPIC='BAR', PARTITIONS=1, VALUE_FORMAT='DELIMITED');"
     );
 
-    // This is needed to make sure that the table is fully done being created.
-    // It's a similar situation to https://github.com/confluentinc/ksql/issues/6249
-    assertThatEventually(
-        () -> makeKsqlQuery("SELECT * FROM migration_schema_versions WHERE VERSION_KEY='CURRENT';").size(),
-        is(1)
-    );
-    final int status = MIGRATIONS_CLI.parse("--config-file", configFilePath, "apply", "-a").run();
-    assertThat(status, is(0));
+    // When:
+    final int applyStatus = MIGRATIONS_CLI.parse("--config-file", configFilePath, "apply", "-a").run();
+
+    // Then:
+    assertThat(applyStatus, is(0));
+
+    verifyMigrationsApplied();
+  }
+
+  private void shouldDisplayInfo() {
+    // Given:
+    Logger.getRootLogger().addAppender(logAppender);
+
+    try {
+      // When:
+      final int infoStatus = MIGRATIONS_CLI.parse("--config-file", configFilePath, "info").run();
+
+      // Then:
+      assertThat(infoStatus, is(0));
+
+      verify(logAppender, atLeastOnce()).doAppend(logCaptor.capture());
+      final List<String> logMessages = logCaptor.getAllValues().stream()
+          .map(LoggingEvent::getRenderedMessage)
+          .collect(Collectors.toList());
+      assertThat(logMessages, hasItem(containsString("Current migration version: 2")));
+      assertThat(logMessages, hasItem(matchesRegex(
+          " Version \\| Name        \\| State    \\| Previous Version \\| Started On\\s+\\| Completed On\\s+\\| Error Reason \n" +
+              "-+\n" +
+              " 1       \\| foo FOO fO0 \\| MIGRATED \\| <none>           \\| \\d{4}-\\d{2}-\\d{2} \\d{2}:\\d{2}:\\d{2}\\.\\d{3} \\S+ \\| \\d{4}-\\d{2}-\\d{2} \\d{2}:\\d{2}:\\d{2}\\.\\d{3} \\S+ \\| N/A          \n" +
+              " 2       \\| bar bar BAR \\| MIGRATED \\| 1                \\| \\d{4}-\\d{2}-\\d{2} \\d{2}:\\d{2}:\\d{2}\\.\\d{3} \\S+ \\| \\d{4}-\\d{2}-\\d{2} \\d{2}:\\d{2}:\\d{2}\\.\\d{3} \\S+ \\| N/A          \n" +
+              "-+\n"
+      )));
+    } finally {
+      Logger.getRootLogger().removeAppender(logAppender);
+    }
+  }
 
+  private static void verifyMigrationsApplied() {
     // verify FOO and BAR were registered
     describeSource("FOO");
     describeSource("BAR");
 
-    // verify current
-    final List<StreamedRow> current = makeKsqlQuery("SELECT * FROM migration_schema_versions WHERE VERSION_KEY='CURRENT';");
-    assertThatEventually(() -> current.size(), is(2));
```

Comment:
These old `assertThatEventually` calls didn't actually do anything since `current` wasn't being refreshed (i.e., `makeKsqlQuery()` was only being called once, outside the `assertThatEventually`). I've gone through and cleaned up the usage of `assertThatEventually` in this test file.